book/forests.qmd

```diff
@@ -41,78 +57,65 @@ $$
+
+Leaves are repeatedly split until a *stopping rule* has been triggered -- i.e., a criterion that tells the algorithm to stop partitioning data.
+The** 'stopping rule' is usually a condition on the number of observations in each leaf such that leaves will continue to be split until some minimum number of observations has been reached in a leaf.
+Other conditions may be on the 'depth' of the tree, which corresponds to the number of levels of splitting, for example the tree in (@fig-surv-ranfor) has a depth of $2$.
+Stopping rules are often used together, for example by setting a maximum tree depth *and* determining a minimum number of observations per leaf.
+Deciding the number of minimum observations and/or the maximum depth can be performed with automated hyper-parameter optimisation.
+
+#### Terminal Node Predictions {.unnumbered .unlisted}
+
+The final major component of decision trees are *terminal node predictions*.
+As the name suggests, this is part of the algorithm that determines how to actually make a prediction for a new observation.
+A prediction is made for some new data by 'dropping' the new data 'down' the tree according to the optimal splits that were found during training.
+Returning to @fig-surv-ranfor, say a new data point is $x = \{`wt` = 3, `disp` = 250\}$, then in the first split the right branch is taken as `wt` $= 3 > 2.32$ and in the second split the left branch is taken as `disp` $= 250 \leq 258$, therefore the new data point 'lands' in the final leaf (the middle gray box).
+
+The resulting prediction is then a simple baseline statistic computed from the training data that fell into the corresponding node.
+In regression, this is most commonly (and simply) the sample mean of the training outcome data, but other terminal node predictions could be the sample median, a weighted average, or even a simple linear regression prediction from a model trained on the training data in the corresponding final leaf.
+In the example above, the final prediction would be $y = 20.755$, note how the final predictions are statistics based on training data, which means all potential predictions can be saved in the original trained mode and no complex computations are required during prediction.
+
+### Random Forests
+
+As is commonly the case in machine learning, higher interpretable yields poor predictive performance.
+Decision trees often overfit the training data, hence they have high variance, perform poorly on new data, and are not robust to even small changes in the original training data.
+Moreover, many important variables can be ignored as only a very small subset are selected, which can be particularly problematic in survival analysis when variables are often know *a priori* to have important clinical value.
```

Comment:
```suggestion
Moreover, important variables can be ignored as only subsets of dominant variables are selected for splitting.
```

Reply:
I think it's not clear here, that this happens because all variables are available for splitting and one the ones with strongest effects are selected for splitting.